qt/scientific_interfaces/EnggDiffraction/EnggVanadiumCorrectionsModel.cpp

```diff
@@ -0,0 +1,190 @@
+      ADS.retrieveWS<Mantid::API::MatrixWorkspace>(CURVES_WORKSPACE_NAME);
+  return std::make_pair(integratedWorkspace, curvesWorkspace);
+}
+
+std::pair<Mantid::API::ITableWorkspace_sptr, Mantid::API::MatrixWorkspace_sptr>
+EnggVanadiumCorrectionsModel::fetchCorrectionWorkspaces(
+    const std::string &vanadiumRunNumber) const {
+  const auto cachedCurvesWorkspace =
+      fetchCachedCurvesWorkspace(vanadiumRunNumber);
+  const auto cachedIntegratedWorkspace =
+      fetchCachedIntegratedWorkspace(vanadiumRunNumber);
+
+  if (cachedCurvesWorkspace && cachedIntegratedWorkspace &&
+      !m_calibSettings.m_forceRecalcOverwrite) {
+    return std::make_pair(*cachedIntegratedWorkspace, *cachedCurvesWorkspace);
+  } else {
+    const auto correctionWorkspaces =
+        calculateCorrectionWorkspaces(vanadiumRunNumber);
+    saveCorrectionsToCache(vanadiumRunNumber, correctionWorkspaces.second,
+                           correctionWorkspaces.first);
+    return correctionWorkspaces;
+  }
+}
+
+boost::optional<Mantid::API::MatrixWorkspace_sptr>
+EnggVanadiumCorrectionsModel::fetchCachedCurvesWorkspace(
+    const std::string &vanadiumRunNumber) const {
+  const auto filename = generateCurvesFilename(vanadiumRunNumber);
+  const Poco::File inputFile(filename);
+  if (inputFile.exists()) {
+    return loadMatrixWorkspace(filename, CURVES_WORKSPACE_NAME);
+  }
+
+  return boost::none;
+}
+
+boost::optional<Mantid::API::ITableWorkspace_sptr>
+EnggVanadiumCorrectionsModel::fetchCachedIntegratedWorkspace(
+    const std::string &vanadiumRunNumber) const {
+  const auto filename = generateIntegratedFilename(vanadiumRunNumber);
+  const Poco::File inputFile(filename);
+  if (inputFile.exists()) {
+    return loadTableWorkspace(filename, INTEGRATED_WORKSPACE_NAME);
+  }
+
+  return boost::none;
+}
+
+std::string EnggVanadiumCorrectionsModel::generateCurvesFilename(
+    const std::string &vanadiumRunNumber) const {
+  const static std::string filenameSuffix =
+      "_precalculated_vanadium_run_bank_curves.nxs";
+
+  const auto normalisedRunName = generateVanadiumRunName(vanadiumRunNumber);
+  const auto curvesFilename = normalisedRunName + filenameSuffix;
+
+  Poco::Path inputDir(m_calibSettings.m_inputDirCalib);
+  inputDir.append(curvesFilename);
+  return inputDir.toString();
+}
+
+std::string EnggVanadiumCorrectionsModel::generateIntegratedFilename(
+    const std::string &vanadiumRunNumber) const {
+  const static std::string filenameSuffix =
+      "_precalculated_vanadium_run_integration.nxs";
+
+  const auto normalisedRunName = generateVanadiumRunName(vanadiumRunNumber);
+  const auto integratedFilename = normalisedRunName + filenameSuffix;
+
+  Poco::Path inputDir(m_calibSettings.m_inputDirCalib);
+  inputDir.append(integratedFilename);
+  return inputDir.toString();
+}
+
+std::string EnggVanadiumCorrectionsModel::generateVanadiumRunName(
+    const std::string &vanadiumRunNumber) const {
+  const static size_t normalisedRunNumberLength = 8;
```

Comment:
How about `constexpr size_t`? Or is this something that could be read from the facility info?